### PR TITLE
add validation to govt inspection sheet

### PR DESCRIPTION
**Pull request by WernerVdM97**
As per request from Heila:

On deletion of a govt inspection pallet, if pallet is repacked, a fail validation is returned to prevent the user from deleting the pallet off of the sheet.

### Changed
- `TaskPermissionCheck::GovtInspectionPallet`, function `delete_check`

### CODE REVIEW

File: lib/finished_goods/task_permission_checks/govt_inspection_pallet.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 module FinishedGoodsApp
   module TaskPermissionCheck
     class GovtInspectionPallet < BaseService
-      attr_reader :task, :entity, :args, :repo
+      attr_reader :task, :id, :entity, :args, :repo
```
Comment (jcsjcs) [inline lib/finished_goods/task_permission_checks/govt_inspection_pallet.rb:6]: There's no need for a reader for `id` - it is never referenced outside the initialize.
In fact the `@id = id` line is redundant.

File: lib/finished_goods/task_permission_checks/govt_inspection_pallet.rb [hunk 2]
```diff
@@ -59,6 +59,7 @@ def edit_check
 
       def delete_check
         return failed_response 'Cant delete Govt Inspection Pallet record if reinspection exists.' if !reinspection? && inspection_count > 1
+        return failed_response 'Cant delete Govt Inspection Pallet if pallet has been repacked.' if repo.get(:pallets, :repacked, repo.get(:govt_inspection_pallets, :pallet_id, id))
```
Comment (jcsjcs) [inline lib/finished_goods/task_permission_checks/govt_inspection_pallet.rb:62]: Please write `Can't` or `Cannot`, but not `Cant`.
Cant: hypocritical and sanctimonious talk, typically of a moral, religious, or political nature.

**Review by jcsjcs — CHANGES_REQUESTED** [review]

File: lib/finished_goods/task_permission_checks/govt_inspection_pallet.rb (hunk 1 above)
Comment (WernerVdM97) [reply lib/finished_goods/task_permission_checks/govt_inspection_pallet.rb:6]: I am referencing id on line 62 at the end of the line to get the pallet id.

Comment (jcsjcs) [reply lib/finished_goods/task_permission_checks/govt_inspection_pallet.rb:6]: OK. missed that.

**Review by jcsjcs — APPROVED** [review]

**Pull request merged**